server/boot/user.js

```diff
@@ -56,6 +55,61 @@ function encodeFcc(value = '') {
   return replaceScriptTags(replaceFormAction(value));
 }
 
+function isAlgorithm(challenge) {
+  // test if name starts with hike/waypoint/basejump/zipline
+  // fix for bug that saved different challenges with incorrect
+  // challenge types
+  return !(/^(waypoint|hike|zipline|basejump)/i).test(challenge.name) &&
```

Comment:
BTW, thinking about this:  When you're doing the data conversion you should also clean up the titles, if you can.  Use this regex (or one like it) to remove these words from them.  Also `checkpoint` which a few people have on theirs.  This will standardize the display names in the profile view.